README.md

```diff
@@ -16,9 +16,125 @@
 ```
 
 ## Install
-```sh
+
+```bash
 npm install @railgun-reloaded/circuits-v2
 ```
 
+## Usage
+
+If you have the params required to call `prove()`, you also have all of the params required to call `verify()`. 
+
+- `prove()` is a more strongly typed and thorough version of what exists in the contracts repo for [forming a proof to prove on-chain](https://github.com/Railgun-Privacy/contract/blob/612b9687eae8c94d34bf09291ec35f1d8eea1ed2/helpers/logic/prover.ts#L49).
+- `verify()` is the typescript version of [proving a proof on-chain](https://github.com/Railgun-Privacy/contract/blob/612b9687eae8c94d34bf09291ec35f1d8eea1ed2/contracts/logic/Verifier.sol#L87)
+
+## Creating new test vectors
```

Comment:
This is excellent documentation. We can leave it here, but personally this is perhaps comments I would have added to the test files such that someone looking at the tests knew where the vectors came from and how to generate them